### ISSUE TITLE
New Rule: Make defaultExport consistent.

### ISSUE THREAD

**Issue by otakutyrant** [eslint/eslint#20147]
### Rule details

When a module default export something, then another module should use the same name when it import the default exported thing.

### Related ECMAScript feature

I think it is default export?

### What type of rule is this?

Warns about a potential problem

### Example code

```js
x.ts:


const a = 1;
export default a;


y.ts:

// This will trigger a warn that `b` is inconsistent the default exported `a` from `x.ts`.
import b from "a";
// This is okay.
import a from "a";
// Explicit rename is okay too.
import a as b from "a";
```

### Why should this rule be in the core instead of a plugin?

Consistence is important. The other benefit may be that when a developer rename default exported name, LS could change all related imported name smartly.

### Participation

- [x] I am willing to submit a pull request to implement this rule.

### Additional comments

I could call AI to write the rule.

**Comment by DMartens** [comment]
Unfortunately ESLint analyzes per file and has no access to the original name of the default export.
The `import` plugin has an export map by parsing the requested file (possibly again).
They have an unresolved [issue](https://github.com/import-js/eslint-plugin-import/issues/1041) for this rule.
